### PR TITLE
Questions

### PR DESCRIPTION
**Pull request by maxmodlol**

### CODE REVIEW

**Comment by Abdallah-Via** [comment]
this pr, should be checked out from master, and contains only 2 files (q1 and a2)
please update it

File: task-json/q2.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const timeinseconds = [4,5,2,10,0];
+timeinseconds.forEach(function  (element){
+    setTimeout(() => {
+        console.log(`print ${element} after ${element} seceonds`);
+
+    },element*1000);
+
```
Comment (Abdallah-Via) [inline task-json/q2.js]: now the code is missy, please note the following:
- create a function called "delay" and take one parameter called seconds
- use a promise to handle the setTimeout
- the numbers will not be printed in the same order, the item (2) will be printed before (5) , what I want is (wait 4 seconds, then print, then wait 5 seconds then ...)

**Review by Abdallah-Via — APPROVED** [review]

File: task-json/q2.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    }).then(function() {
+       
+         console.log(`${seconds} seconds` );
+        
+       
+    });
+}
+for(let x=0 ;x<timeinseconds.length;x++){
+    delay(timeinseconds[x]);
+}
```
Comment (Abdallah-Via) [inline task-json/q2.js:27]: Did you test the results?

File: task-json/q3.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+function generateArrayOfNumbers(arrayLength){
+    const arr=new Array(arrayLength);
+    
+
+arr.fill().map((value,indx)=>(arr[indx]=Math.floor(Math.random()*10)));
```
Comment (Abdallah-Via) [inline task-json/q3.js]: use it using normal for loop

File: task-json/q3.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+arr.fill().map((value,indx)=>(arr[indx]=Math.floor(Math.random()*10)));
+
+    return arr ;
+}
+const arr=generateArrayOfNumbers(100);
+function getArrayOfNumbersWithoutDuplication(arrayOfNumbers) {
+    unique = new Set(arrayOfNumbers);
+    console.log(unique);
+    return unique
```
Comment (Abdallah-Via) [inline task-json/q3.js]: this will not return array

File: task-json/q3.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+function getArrayOfNumbersWithoutDuplication1(arrayOfNumbers) {
+
+uniqueArray = arrayOfNumbers.filter(function(item, pos) {
+    return arrayOfNumbers.indexOf(item) == pos;
+})
+
+console.log(uniqueArray);
+
+return uniqueArray;
+}
```
Comment (Abdallah-Via) [inline task-json/q3.js:24]: Please run your code locally :)
this code will not work

File: task-json/q3.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+function getDuplication(arrayOfNumbers) {
+    const dublicte= new Map(); 
+    for(let i =0;i<arrayOfNumbers.length;i++){
+        dublicte.set(arrayOfNumbers[i],dublicte.get(arrayOfNumbers[i]) + 1 || 1);
+
```
Comment (Abdallah-Via) [inline task-json/q3.js:29]: why || 1

File: task-json/q3.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+uniqueArray = arrayOfNumbers.filter(function(item, pos) {
+    return arrayOfNumbers.indexOf(item) == pos;
+})
+
+console.log(uniqueArray);
+
+return uniqueArray;
+}
+function getDuplication(arrayOfNumbers) {
+    const dublicte= new Map();
```
Comment (Abdallah-Via) [inline task-json/q3.js:26]: this function will return array of duplicated numbers, not their count

File: task-json/q3.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    for(let i =0;i<arrayOfNumbers.length;i++){
+        dublicte.set(arrayOfNumbers[i],dublicte.get(arrayOfNumbers[i]) + 1 || 1);
+
+    } 
+    console.log(dublicte);
+   return dublicte
+}
+
+function getDuplication2(arrayOfNumbers) {
+
```
Comment (Abdallah-Via) [inline task-json/q3.js:36]: this function will return array of duplicated numbers, not their count

File: task-json/q2.js (hunk 2 above)
Comment (maxmodlol) [reply task-json/q2.js:27]: yes its not working as it should be its printing the numbers in the same order  in the array but the timeout function is not responding

File: task-json/q3.js (hunk 3 above)
Comment (maxmodlol) [reply task-json/q3.js]: u mean not using fill and map instead i use for loop ?

File: task-json/q3.js (hunk 6 above)
Comment (maxmodlol) [reply task-json/q3.js:29]: because if there is a value just add a +1 on it if there is not a value assign it 1

File: task-json/q3.js (hunk 5 above)
Comment (maxmodlol) [reply task-json/q3.js:24]: its working i just test it out !

File: task-json/q3.js (hunk 7 above)
Comment (maxmodlol) [reply task-json/q3.js:26]: it will return a map with key and value 
key is the number and value is their count

File: task-json/q3.js (hunk 8 above)
Comment (maxmodlol) [reply task-json/q3.js:36]: its  returning array of key and value

**Pull request merged**